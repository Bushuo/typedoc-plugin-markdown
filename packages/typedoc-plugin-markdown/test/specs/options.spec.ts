import * as Handlebars from 'handlebars';
import { TestApp } from '../test-app';

describe(`Options:`, () => {
  let testApp: TestApp;

  beforeAll(() => {
    testApp = new TestApp(['breadcrumbs.ts']);
  });

  describe(`(defaults)`, () => {
    beforeAll(async () => {
      await testApp.bootstrap({});
    });

    test(`should set 'ifShowNamedAnchors' to 'false' by default`, () => {
      expect(
        Handlebars.helpers.ifShowNamedAnchors(TestApp.handlebarsOptionsStub),
      ).toBeFalsy();
    });

    test(`should set 'ifShowBreadcrumbs' to 'true' by default`, () => {
      expect(
        Handlebars.helpers.ifShowBreadcrumbs(TestApp.handlebarsOptionsStub),
      ).toBeTruthy();
    });

    test(`should set 'ifShowReflectionPath' to 'false' by default`, () => {
      expect(
        Handlebars.helpers.ifShowReflectionPath(TestApp.handlebarsOptionsStub),
      ).toBeFalsy();
    });

    test(`should set 'ifShowIndex' to 'true' by default`, () => {
      expect(
        Handlebars.helpers.ifShowIndex(TestApp.handlebarsOptionsStub),
      ).toBeTruthy();
    });

    test(`should compile relativeURL helper`, () => {
      const url = testApp.findReflection('Breadcrumbs').url;
      expect(Handlebars.helpers.relativeURL(url)).toMatchSnapshot();
    });
  });

  describe(`(with plugin options)`, () => {
    beforeAll(async () => {
      await testApp.bootstrap({
        publicPath: 'test-public-path',
        namedAnchors: true,
        hideBreadcrumbs: true,
        navigationEnabled: true,
      });
    });

    test(`should set named anchors to 'true'`, () => {
      expect(
        Handlebars.helpers.ifShowNamedAnchors(TestApp.handlebarsOptionsStub),
      ).toBeTruthy();
    });

    test(`should set breadcrumbs to 'false'`, () => {
      expect(
        Handlebars.helpers.ifShowBreadcrumbs(TestApp.handlebarsOptionsStub),
      ).toBeFalsy();
    });

    test(`should compile relativeURL helper with public path`, () => {
      const url = testApp.findReflection('Breadcrumbs').url;
      expect(Handlebars.helpers.relativeURL(url)).toMatchSnapshot();
    });

    test(`should set 'ifShowReflectionPath' to true if navigation enabled`, () => {
      expect(
        Handlebars.helpers.ifShowReflectionPath(TestApp.handlebarsOptionsStub),
      ).toBeTruthy();
    });

    test(`should set 'ifShowIndex' to false on classes if navigation enabled`, () => {
      const reflection = testApp.findReflection('Breadcrumbs');
      expect(
        Handlebars.helpers.ifShowIndex.call(
          reflection,
          TestApp.handlebarsOptionsStub,
        ),
      ).toBeFalsy();
    });

    test(`should set 'ifShowIndex' to true on modules if navigation enabled`, () => {
      expect(
        Handlebars.helpers.ifShowIndex(TestApp.handlebarsOptionsStub),
      ).toBeTruthy();
    });
  });
});
