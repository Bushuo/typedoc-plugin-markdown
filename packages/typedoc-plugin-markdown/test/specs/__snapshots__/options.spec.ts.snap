// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options: (defaults) should compile relativeURL helper 1`] = `"breadcrumbs.md"`;

exports[`Options: (with plugin options) should compile relativeURL helper with public path 1`] = `"test-public-pathclasses/breadcrumbs.md"`;
